File: src/app/models/LdapNode.ts
export interface LdapNode {
	id : string;
	label : string;
	toggled: boolean;
	hasChildren: boolean;
	dn: string;                   // Distinguished Name (DN)
	attributes?: Record<string, string[]>;         // List of attributes associated with the entry (multi-valued)
	children?: LdapNode[];         // List of child LdapNode objects (recursive structure)
}

File: src-tauri/src/models/ldap_node.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LdapNode {
    pub dn: String,                // Distinguished Name
    pub attributes: Vec<String>,   // Attributes of the entry
    pub children: Vec<LdapNode>,   // Children of the entry
    pub toggled : bool,
    pub has_children: bool
}
